### PR TITLE
don't use SDL_RenderSetLogicalSize

### PR DESCRIPTION
**Pull request by rfomin**
Alternative to #1844

### CODE REVIEW

File: src/i_video.c [hunk 1]
```diff
@@ -662,7 +667,6 @@ static void UpdateRender(void)
     SDL_LowerBlit(screenbuffer, &blit_rect, argbbuffer, &blit_rect);
     SDL_UpdateTexture(texture, &blit_rect, argbbuffer->pixels,
                       argbbuffer->pitch);
-    SDL_RenderClear(renderer);
```
Comment (gendlin) [inline src/i_video.c:665]: Without this clear for pillar/letterbox cases stuff like RTSS that draws into the framebuffer will get messed up, and `vga_porch_flash` won't work. Other than that this patch seems fine

Comment (rfomin) [reply src/i_video.c:665]: I fixed vga_porch here: https://github.com/fabiangreffrath/woof/pull/1847/commits/6380fa33c370dad2e321c29ef0600cd6c1b5c83a

What if we just call `SDL_RenderClear(renderer)` unconditionally? Seems safer, the GPU is idle most of the time anyway.

Comment (gendlin) [reply src/i_video.c:665]: It's just useless work that can be avoided with a simple flag check 🤷‍♂️

Comment (rfomin) [reply src/i_video.c:665]: Okay, fixed here: https://github.com/fabiangreffrath/woof/pull/1847/commits/5f63d4cfcbb3f0c674b724ff750902082a461743

**Comment by fabiangreffrath** [comment]
Looks clean, but in how far is this better than just calling `SDL_RenderSetLogicalSize()`?

**Comment by rfomin** [comment]
> Looks clean, but in how far is this better than just calling `SDL_RenderSetLogicalSize()`?

This should fix the thin black bars described here: https://github.com/fabiangreffrath/woof/pull/1844#issuecomment-2294794716

**Comment by fabiangreffrath** [comment]
So this fixes #1601 ?

File: src/i_video.c [hunk 2]
```diff
@@ -1389,15 +1399,53 @@ static void CreateUpscaledTexture(boolean force)
...
+        viewport.w = w;
+        viewport.h = h;
+        letterboxed = false;
+    }
+    else if (want_aspect > real_aspect)
+    {
+        viewport.w = w;
+        viewport.h = (int)floor(actualheight * scalex);
+        viewport.y = (h - viewport.h) / 2;
+        letterboxed = true;
```
Comment (gendlin) [inline src/i_video.c:1429]:
```c
scaley = scalex;
```

File: src/i_video.c [hunk 3]
```diff
@@ -1389,15 +1399,53 @@ static void CreateUpscaledTexture(boolean force)
...
+    }
+    else
+    {
+        viewport.h = h;
+        viewport.w = (int)floor(video.width * scaley);
+        viewport.x = (w - viewport.w) / 2;
+        letterboxed = true;
```
Comment (gendlin) [inline src/i_video.c:1436]:
```c
scalex = scaley;
```

**Review by gendlin — CHANGES_REQUESTED** [review]

**Comment by rfomin** [comment]
> So this fixes #1601 ?

No, that HOM is an issue with our internal resolution/rendering. This fixes a similar problem with SDL scaling on the GPU. Try this diff with a master branch:
```diff
diff --git a/src/i_video.c b/src/i_video.c
index 9823b7a3..7ec7efc4 100644
--- a/src/i_video.c
+++ b/src/i_video.c
@@ -659,6 +659,7 @@ void I_StartFrame(void)
 
 static void UpdateRender(void)
 {
+    SDL_SetRenderDrawColor(renderer, 255, 0, 0, SDL_ALPHA_OPAQUE);
     SDL_LowerBlit(screenbuffer, &blit_rect, argbbuffer, &blit_rect);
     SDL_UpdateTexture(texture, &blit_rect, argbbuffer->pixels,
                       argbbuffer->pitch);
```

File: src/i_video.c (hunk 2 above)
Comment (rfomin) [reply src/i_video.c:1429]: This is from the SDL code, see https://github.com/libsdl-org/SDL/blob/a3546e50697f2424a39f991148708c60903c553b/src/render/SDL_render.c#L2391

Comment (gendlin) [reply src/i_video.c:1429]: The scale must be uniform here. It's the same in SDL: https://github.com/libsdl-org/SDL/blob/a3546e50697f2424a39f991148708c60903c553b/src/render/SDL_render.c#L2429

Comment (rfomin) [reply src/i_video.c:1430]: Fixed here: https://github.com/fabiangreffrath/woof/pull/1847/commits/83d3e0af3bdadf6368f6a3cb50168efb63884ec5

File: src/i_video.c [hunk 4]
```diff
@@ -1417,27 +1416,30 @@ static void UpdateViewport(void)
 
     if (fabs(want_aspect - real_aspect) < 0.0001)
     {
+        scale = (float)w / video.width;
```
Comment (gendlin) [inline src/i_video.c:1419]: The scales should differ here, we are stretching to cover the window

File: src/i_video.c (hunk 2 above)
Comment (gendlin) [reply src/i_video.c:1430]: First case should not be modified, your original code + uniform scale in letterboxed cases is correct approach

File: src/i_video.c (hunk 4 above)
Comment (rfomin) [reply src/i_video.c]: Okay, how about this: https://github.com/fabiangreffrath/woof/pull/1847/commits/6aabb3f181044583f76a675d7d9398f5bcc4fa0c

Interestingly, all of the variants work either way 😄

File: src/i_video.c (hunk 2 above)
Comment (rfomin) [reply src/i_video.c]: So it's all correct now? 6aabb3f
If not, post your diff or the entire `UpdateViewport` function here.

Comment (gendlin) [reply src/i_video.c]: Looks good, thanks

**Pull request merged**